test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SpanExpectation.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.TestHelpers;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    /// <summary>
+    /// Base class for all span expectations. Inherit from this class to extend per type or integration.
+    /// </summary>
+    public class SpanExpectation
+    {
+        public SpanExpectation(string serviceName, string operationName, string type)
+        {
+            ServiceName = serviceName;
+            OperationName = operationName;
+            Type = type;
+
+            // Expectations for all spans regardless of type should go here
+            RegisterDelegateExpectation(ExpectBasicSpanDataExists);
+
+            RegisterCustomExpectation(nameof(OperationName), actual: s => s.Name, expected => OperationName);
+            RegisterCustomExpectation(nameof(ServiceName), actual: s => s.Service, expected => ServiceName);
+            RegisterCustomExpectation(nameof(Type), actual: s => s.Type, expected => Type);
+            RegisterCustomExpectation(nameof(ResourceName), actual: s => s.Resource.TrimEnd(), expected => ResourceName);
+
+            RegisterTagExpectation(
+                key: nameof(Tags.Language),
+                expected: TracerConstants.Language,
+                when: s => GetTag(s, Tags.SpanKind) != SpanKinds.Client);
+        }
+
+        public Func<MockTracerAgent.Span, bool> Always => s => true;
+
+        public List<Func<MockTracerAgent.Span, string>> Assertions { get; } = new List<Func<MockTracerAgent.Span, string>>();
+
+        public bool IsTopLevel { get; set; } = true;
+
+        public string Type { get; set; }
+
+        public string ResourceName { get; set; }
+
+        public string OperationName { get; set; }
+
+        public string ServiceName { get; set; }
+
+        public virtual string Detail()
+        {
+            return $"service={ServiceName}, operation={OperationName}, type={Type}, resource={ResourceName}";
+        }
+
+        /// <summary>
+        /// Override for custom filters.
+        /// </summary>
+        /// <param name="span">The span on which to filter.</param>
+        /// <returns>Whether the span qualifies for this expectation.</returns>
+        public virtual bool ShouldInspect(MockTracerAgent.Span span)
```

Comment:
I feel like another verb would help make this a little bit clearer when reading over SpanTestHelpers.cs. Maybe `MatchesSpan`?

Reply:
How about just `Matches(span)`?

Reply:
Yeah that sounds good to me 👍